src/particles/diagnostics/DiagnosticOutput.H

```diff
@@ -42,7 +44,9 @@ namespace impactx::diagnostics
                            OutputType otype,
```

Comment:
My feeling is we could make this an overload of `DiagnosticOutput`, because we ignore either `pc` or `cm` currently when we call it.

Reply:
This I'm not sure I understand.  We never ignore `pc`, because we always need the reference particle information.

Reply:
I refactored a bit more and merged.

The `ref` particle we can track separately in envelope mode, without attaching it to a particle container.